packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -199,11 +206,36 @@ const PostsPage = ({post, refetch, classes}: {
     skip: !post.question,
   });
 
+  const { results: debateComments } = useMulti({
+    terms: {
+      view: 'debateComments',
+      postId: post._id,
+    },
+    collectionName: 'Comments',
+    fragmentName: 'CommentsList',
+    skip: !post.debate,
+    limit: 1000
+  });
+
+  const defaultView = commentGetDefaultView(post, currentUser)
+  // If the provided view is among the valid ones, spread whole query into terms, otherwise just do the default query
+  const commentTerms: CommentsViewTerms = Object.keys(viewNames).includes(query.view)
+    ? {...(query as CommentsViewTerms), limit:1000}
+    : {view: defaultView, limit: 1000}
+
+  const { results: nonDebateComments } = useMulti({
+    terms: {...commentTerms, postId: post._id},
+    collectionName: "Comments",
+    fragmentName: 'CommentsList',
+    fetchPolicy: 'cache-and-network',
+    skip: !post.debate
+  });
```

Comment:
Consider extracting into a hook for ease of reading this component.

Reply:
potentially worth follow-up